edgelet/iotedged/src/app.rs

```diff
@@ -25,81 +25,67 @@ pub fn create_base_app<'a, 'b>() -> App<'a, 'b> {
                 .value_name("FILE")
                 .help("Sets daemon configuration file")
                 .takes_value(true),
+        );
+
+    if cfg!(windows) {
+        app.arg(
+            Arg::with_name("use-event-logger")
+                .short("e")
+                .long("use-event-logger")
+                .value_name("USE_EVENT_LOGGER")
+                .help("Log to Windows event logger instead of stdout")
+                .required(false)
+                .takes_value(false),
         )
+    } else {
+        app
+    }
 }
 
-#[cfg(not(target_os = "windows"))]
-pub fn create_app<'a, 'b>() -> App<'a, 'b> {
-    create_base_app()
-}
+fn init_common(running_as_windows_service: bool) -> Result<Settings<DockerConfig>, Error> {
+    let matches = create_app().get_matches();
 
-#[cfg(target_os = "windows")]
-pub fn create_app<'a, 'b>() -> App<'a, 'b> {
-    create_base_app().arg(
-        Arg::with_name("use-event-logger")
-            .short("e")
-            .long("use-event-logger")
-            .value_name("USE_EVENT_LOGGER")
-            .help("Log to Windows event logger instead of stdout")
-            .required(false)
-            .takes_value(false),
-    )
-}
+    // If running as a Windows service, logging was already initialized by init_win_svc_logging(), so don't do it again.
+    if !running_as_windows_service {
```

Comment:
Would this be cleaner as:
```
#[cfg(windows)]
if matches.is_present(...) { 
   ... 
}
#cfg(not(windows))
logging::init()
```

Reply:
`logging::init()` is not specific to non-Windows.